cmds/watch.c

```diff
@@ -27,25 +27,43 @@ bool w_cmd_realpath_root(json_t *args, char **errmsg)
   return true;
 }
 
-/* clock /root
+/* clock /root [options]
  * Returns the current clock value for a watched root
+ * If the options contain a sync_timeout, we ensure that the repo
+ * is synced up-to-date and the returned clock represents the
+ * latest state.
  */
 static void cmd_clock(struct watchman_client *client, json_t *args)
 {
   w_root_t *root;
   json_t *resp;
-
-  /* resolve the root */
-  if (json_array_size(args) != 2) {
+  unsigned int sync_timeout = 0;
+
+  if (json_array_size(args) == 3) {
+    json_t *options = json_array_get(args, 2);
+    if (options) {
+      json_t *st_json = json_object_get(options, "sync_timeout");
+      if (st_json) {
+        sync_timeout = (unsigned int)json_integer_value(st_json);
+      }
+    }
```

Comment:
you can use something like:

```
json_unpack(args, "[s, s, {s?:d}]", &ignored, &ignored, "sync_timeout", &sync_timeout);
```

to express this a bit more tersely.  Take a look at `cmd_version` in `cmds/info.c` for an example